plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/githubActionsDispatch.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  GithubCredentialsProvider,
+  ScmIntegrationRegistry,
+} from '@backstage/integration';
+import { Octokit } from '@octokit/rest';
+import { parseRepoUrl } from '../publish/util';
+import { createTemplateAction } from '../../createTemplateAction';
+
+export function createGithubActionsDispatchAction(options: {
+  integrations: ScmIntegrationRegistry;
+}) {
+  const { integrations } = options;
+
+  const credentialsProviders = new Map(
+    integrations.github.list().map(integration => {
+      const provider = GithubCredentialsProvider.create(integration.config);
+      return [integration.config.host, provider];
+    }),
+  );
+
+  return createTemplateAction<{
+    repoUrl: string;
+    workflowId: string;
+    branchOrTagName: string;
+  }>({
+    id: 'github:actions:dispatch',
+    description:
+      'Dispatches a GitHub Action workflow for a given branch or tag',
+    schema: {
+      input: {
+        type: 'object',
+        required: ['repoUrl', 'workflowId', 'branchOrTagName'],
+        properties: {
+          repoUrl: {
+            title: 'Repository Location',
+            description: `Accepts the format 'github.com?repo=reponame&owner=owner' where 'reponame' is the new repository name and 'owner' is an organization or username`,
+            type: 'string',
+          },
+          workflowId: {
+            title: 'Workflow ID',
+            description: 'The GitHub Action Workflow filename',
+            type: 'string',
+          },
+          branchOrTagName: {
+            title: 'Branch or Tag name',
+            description:
+              'The git branch or tag name used to dispatch the workflow',
+            type: 'string',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const { repoUrl, workflowId, branchOrTagName } = ctx.input;
+
+      const { owner, repo, host } = parseRepoUrl(repoUrl);
+
+      ctx.logger.info(
+        `Dispatching workflow ${workflowId} for repo ${repoUrl} on ${branchOrTagName}`,
+      );
+
+      const credentialsProvider = credentialsProviders.get(host);
+      const integrationConfig = integrations.github.byHost(host);
+
+      if (!credentialsProvider || !integrationConfig) {
+        throw new InputError(
+          `No matching integration configuration for host ${host}, please check your integrations config`,
+        );
+      }
+
+      const { token } = await credentialsProvider.getCredentials({
+        url: `https://${host}/${encodeURIComponent(owner)}/${encodeURIComponent(
+          repo,
+        )}`,
+      });
+
+      if (!token) {
+        throw new InputError(
+          `No token available for host: ${host}, with owner ${owner}, and repo ${repo}`,
+        );
+      }
+
+      const client = new Octokit({
+        auth: token,
+        baseUrl: integrationConfig.config.apiBaseUrl,
+        previews: ['nebula-preview'],
+      });
+
+      await client.rest.actions.createWorkflowDispatch({
```

Comment:
I wonder if it also makes sense to define outputs here, like the run URL or something. Don't need to do it in this PR, but it will be useful if you want to show that run link to the end user too.